src/common/abstractPromote.ts

```diff
@@ -101,6 +106,23 @@ export abstract class PromoteCommand<T extends typeof SfCommand> extends SfComma
     };
   }
 
+  protected async retryEnabledRequestPromotion(doceOrg: Org, numRetries: number): Promise<string> {
+    try {
+      const asyncOperationId: string = await this.requestPromotion(doceOrg);
+      return asyncOperationId;
+    } catch (error) {
+      const err = error as HttpResponse;
+      // if we get anything besdies a 409 or run out of retries we throw an error
+      if (err['errorCode'] !== 'ERROR_HTTP_409' || numRetries < 1) {
+        return JSON.stringify(err); // we can't throw the error from this promise because then it doesn't surface to the user
```

Comment:
Wouldn't it be better to change the return type of this method to `Promise<string|HttpResponse>` ?  Then the caller would have full access to the error (and could rethrow it if needed)